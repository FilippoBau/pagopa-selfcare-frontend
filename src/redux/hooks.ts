import {TypedUseSelectorHook, useDispatch, useSelector} from 'react-redux';
import {useHistory} from 'react-router-dom';
import type {RootState, AppDispatch} from './store';

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

<<<<<<< HEAD
export const useAppSelectorWithRedirect = ({
  selector,
  routeToRedirect,
  conditionToRedirect,
}: {
  selector: (state: RootState) => any;
  routeToRedirect?: string;
  conditionToRedirect?: (value: any) => boolean;
}) => {
  const selectedReduxState = useAppSelector(selector);
  const history = useHistory();
  const conditionResults = conditionToRedirect ? conditionToRedirect(selectedReduxState) : false;
  if (
    routeToRedirect &&
    (!selectedReduxState ||
      (typeof selectedReduxState === 'object' && Object.keys(selectedReduxState).length === 0) ||
      conditionResults)
  ) {
    history.push(routeToRedirect);
  }
  return selectedReduxState;
=======
export const useAppSelectorWithRedirect = (
    selector: (state: RootState) => any,
    routeToRedirect?: string
) => {
    const selectedReduxState = useAppSelector(selector);
    const history = useHistory();
    if (
        routeToRedirect &&
        (!selectedReduxState ||
            (typeof selectedReduxState === 'object' && Object.keys(selectedReduxState).length === 0))
    ) {
        history.push(routeToRedirect);
    }
    return selectedReduxState;
>>>>>>> 3f32cfc3 (Formatting (#542))
};
