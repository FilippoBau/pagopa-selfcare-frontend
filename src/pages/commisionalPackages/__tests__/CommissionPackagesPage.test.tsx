import { ThemeProvider } from '@mui/system';
import { theme } from '@pagopa/mui-italia';
import { render } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import { store } from '../../../redux/store';
import { Provider } from 'react-redux';
import React from 'react';
import CommissionPackagesPage from '../CommissionPackagesPage';

describe('<CommissionPackagesPage />', () => {
  test('render component CommissionPackagesPage', async () => {
    render(
      <Provider store={store}>
        <MemoryRouter initialEntries={[`/comm-packages`]}>
          <Route path="/comm-packages">
            <ThemeProvider theme={theme}>
              <CommissionPackagesPage />
            </ThemeProvider>
          </Route>
        </MemoryRouter>
      </Provider>
    );
  });
});
