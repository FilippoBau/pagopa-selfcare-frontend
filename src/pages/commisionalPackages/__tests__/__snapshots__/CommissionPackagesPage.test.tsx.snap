// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-4uxo5g-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1539t99-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div
              class="MuiBox-root css-jeqi4w"
            >
              <div
                class="MuiBox-root css-ir1t2y"
              >
                <ul
                  class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                >
                  <div
                    class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1k5qf7y-MuiButtonBase-root-MuiListItemButton-root"
                    data-testid="home-test"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiListItemIcon-root css-1vo01mv-MuiListItemIcon-root"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-ycncy1-MuiSvgIcon-root-MuiIcon-root"
                        data-testid="DashboardIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiListItemText-root css-mmjwmt-MuiListItemText-root"
                    >
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1gq3qqz-MuiTypography-root"
                      >
                        sideMenu.home.title
                      </span>
                    </div>
                  </div>
                </ul>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-13qir85-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-v4b62z"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-14b6npa-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-13e0yuj-MuiGrid-root"
              >
                <h4
                  class="MuiTypography-root MuiTypography-h4 css-tp226b-MuiTypography-root"
                >
                  commissionPackagesPage.title
                </h4>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-kxmh4q-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1mx152z-MuiTypography-root"
                >
                  commissionPackagesPage.subtitle
                </p>
              </div>
            </div>
            <div
              class="MuiBox-root css-18p9amc"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-g598ww-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-i29xuw-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiOutlinedInput-input MuiInputBase-input css-y2nd8p-MuiInputBase-input-MuiOutlinedInput-input"
                    data-testid="search-name-test"
                    id="mui-1"
                    placeholder="commissionPackagesPage.list.searchByName"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-hdw1oc"
                    >
                      <span
                        class="notranslate"
                      >
                        â€‹
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <a
                class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1pa4f95-MuiButtonBase-root-MuiButton-root"
                href="/ui/comm-packages/add-package/"
                tabindex="0"
              >
                commissionPackagesPage.list.createPackage
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
            </div>
            <div
              class="MuiBox-root css-suu0r8"
            >
              <div
                class="MuiTabs-root css-1uzak9y-MuiTabs-root"
              >
                <div
                  class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
                  style="overflow: hidden; margin-bottom: 0px;"
                >
                  <div
                    class="MuiTabs-flexContainer MuiTabs-centered css-1wf8b0h-MuiTabs-flexContainer"
                    role="tablist"
                  >
                    <button
                      aria-controls="simple-tabpanel-0"
                      aria-selected="false"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth css-1trwt52-MuiButtonBase-root-MuiTab-root"
                      id="simple-tab-0"
                      role="tab"
                      tabindex="-1"
                      type="button"
                    >
                      commissionPackagesPage.globalPackages
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-controls="simple-tabpanel-1"
                      aria-selected="false"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth css-1trwt52-MuiButtonBase-root-MuiTab-root"
                      id="simple-tab-1"
                      role="tab"
                      tabindex="-1"
                      type="button"
                    >
                      commissionPackagesPage.publicPackages
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                    <button
                      aria-controls="simple-tabpanel-2"
                      aria-selected="true"
                      class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected css-1trwt52-MuiButtonBase-root-MuiTab-root"
                      id="simple-tab-2"
                      role="tab"
                      tabindex="0"
                      type="button"
                    >
                      commissionPackagesPage.privatePackages
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <span
                    class="MuiTabs-indicator css-axdyhn-MuiTabs-indicator"
                    style="left: 0px; width: 0px;"
                  />
                </div>
              </div>
            </div>
            <div
              aria-labelledby="simple-tab-0"
              hidden=""
              id="simple-tabpanel-0"
              role="tabpanel"
            />
            <div
              aria-labelledby="simple-tab-1"
              hidden=""
              id="simple-tabpanel-1"
              role="tabpanel"
            />
            <div
              aria-labelledby="simple-tab-2"
              id="simple-tabpanel-2"
              role="tabpanel"
            >
              <div
                class="MuiBox-root css-17yyzkl"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-11o4oo8-MuiTypography-root"
                >
                  <div
                    class="MuiBox-root css-1cg1tk5"
                    id="commissionPackagesTable"
                  >
                    <div
                      class="MuiBox-root css-1fwt4j5"
                    >
                      <div
                        class="MuiBox-root css-1podeb3"
                      >
                        <p
                          class="MuiTypography-root MuiTypography-body2 css-pjo5ov-MuiTypography-root"
                        >
                          Non sono ancora presenti pacchetti commissioni 
                          commissionPackagesPage.privatePackages
                          .
                          <a
                            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1lr2oxb-MuiTypography-root-MuiLink-root"
                            href="/ui/comm-packages/add-package/"
                          >
                            <strong>
                               Crea Pacchetto
                            </strong>
                          </a>
                        </p>
                      </div>
                    </div>
                  </div>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-4uxo5g-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1539t99-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-jeqi4w"
          >
            <div
              class="MuiBox-root css-ir1t2y"
            >
              <ul
                class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
              >
                <div
                  class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-1k5qf7y-MuiButtonBase-root-MuiListItemButton-root"
                  data-testid="home-test"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemIcon-root css-1vo01mv-MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-ycncy1-MuiSvgIcon-root-MuiIcon-root"
                      data-testid="DashboardIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                      />
                    </svg>
                  </div>
                  <div
                    class="MuiListItemText-root css-mmjwmt-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-1gq3qqz-MuiTypography-root"
                    >
                      sideMenu.home.title
                    </span>
                  </div>
                </div>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-13qir85-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-v4b62z"
        >
          <div
            class="MuiGrid-root MuiGrid-container css-14b6npa-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-13e0yuj-MuiGrid-root"
            >
              <h4
                class="MuiTypography-root MuiTypography-h4 css-tp226b-MuiTypography-root"
              >
                commissionPackagesPage.title
              </h4>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-kxmh4q-MuiGrid-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-1mx152z-MuiTypography-root"
              >
                commissionPackagesPage.subtitle
              </p>
            </div>
          </div>
          <div
            class="MuiBox-root css-18p9amc"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-g598ww-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-i29xuw-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-y2nd8p-MuiInputBase-input-MuiOutlinedInput-input"
                  data-testid="search-name-test"
                  id="mui-1"
                  placeholder="commissionPackagesPage.list.searchByName"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-hdw1oc"
                  >
                    <span
                      class="notranslate"
                    >
                      â€‹
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <a
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1pa4f95-MuiButtonBase-root-MuiButton-root"
              href="/ui/comm-packages/add-package/"
              tabindex="0"
            >
              commissionPackagesPage.list.createPackage
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
          <div
            class="MuiBox-root css-suu0r8"
          >
            <div
              class="MuiTabs-root css-1uzak9y-MuiTabs-root"
            >
              <div
                class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
                style="overflow: hidden; margin-bottom: 0px;"
              >
                <div
                  class="MuiTabs-flexContainer MuiTabs-centered css-1wf8b0h-MuiTabs-flexContainer"
                  role="tablist"
                >
                  <button
                    aria-controls="simple-tabpanel-0"
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth css-1trwt52-MuiButtonBase-root-MuiTab-root"
                    id="simple-tab-0"
                    role="tab"
                    tabindex="-1"
                    type="button"
                  >
                    commissionPackagesPage.globalPackages
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-controls="simple-tabpanel-1"
                    aria-selected="false"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth css-1trwt52-MuiButtonBase-root-MuiTab-root"
                    id="simple-tab-1"
                    role="tab"
                    tabindex="-1"
                    type="button"
                  >
                    commissionPackagesPage.publicPackages
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                  <button
                    aria-controls="simple-tabpanel-2"
                    aria-selected="true"
                    class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected css-1trwt52-MuiButtonBase-root-MuiTab-root"
                    id="simple-tab-2"
                    role="tab"
                    tabindex="0"
                    type="button"
                  >
                    commissionPackagesPage.privatePackages
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <span
                  class="MuiTabs-indicator css-axdyhn-MuiTabs-indicator"
                  style="left: 0px; width: 0px;"
                />
              </div>
            </div>
          </div>
          <div
            aria-labelledby="simple-tab-0"
            hidden=""
            id="simple-tabpanel-0"
            role="tabpanel"
          />
          <div
            aria-labelledby="simple-tab-1"
            hidden=""
            id="simple-tabpanel-1"
            role="tabpanel"
          />
          <div
            aria-labelledby="simple-tab-2"
            id="simple-tabpanel-2"
            role="tabpanel"
          >
            <div
              class="MuiBox-root css-17yyzkl"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-11o4oo8-MuiTypography-root"
              >
                <div
                  class="MuiBox-root css-1cg1tk5"
                  id="commissionPackagesTable"
                >
                  <div
                    class="MuiBox-root css-1fwt4j5"
                  >
                    <div
                      class="MuiBox-root css-1podeb3"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2 css-pjo5ov-MuiTypography-root"
                      >
                        Non sono ancora presenti pacchetti commissioni 
                        commissionPackagesPage.privatePackages
                        .
                        <a
                          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1lr2oxb-MuiTypography-root-MuiLink-root"
                          href="/ui/comm-packages/add-package/"
                        >
                          <strong>
                             Crea Pacchetto
                          </strong>
                        </a>
                      </p>
                    </div>
                  </div>
                </div>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
